scripts/simple-scylla-cluster-manager.py

```diff
@@ -0,0 +1,776 @@
+            return self._make_fail(f"Failed to find node#{i}")
+
+        if i in self._nodes:
+            return self._make_success(f"Node#{i} is already running")
+
+        node_config = self._config["nodes"][i]
+
+        node_path = os.path.abspath(f"node{i}")
+        node_smp = self._get_node_config_item(i, "smp")
+        node_memory = self._get_node_config_item(i, "memory")
+        node_ip = node_config["ip"]
+
+        if self._nodes:
+            min_node_index = min(self._nodes.keys())
+            seed_node_ip = self._nodes[min_node_index]["ip"]
+        else:
+            seed_node_ip = node_ip
+
+        invoke_cmd = [os.path.join(node_path, "scylla"),
+                f"-c{node_smp}",
+                f"-m{node_memory}",
+                "--cluster-name", self._config['name'],
+                "--options-file", os.path.join(node_path, "scylla.yaml"),
+                "--developer-mode=1",
+                "--overprovisioned",
+                "--unsafe-bypass-fsync=1",
+                "--relaxed-dma",
+                "--listen-address", node_ip,
+                "--rpc-address", node_ip,
+                "--api-address", node_ip,
+                "--prometheus-address", node_ip,
+                "--seed-provider-parameters", f"seeds={seed_node_ip}",
+                "--data-file-directories", os.path.join(node_path, "data"),
+                "--commitlog-directory", os.path.join(node_path, "commitlog"),
+                "--hints-directory", os.path.join(node_path, "hints"),
+                "--view-hints-directory", os.path.join(node_path, "view_hints"),
+                "--api-ui-dir", os.path.join(node_path, "api-ui"),
+                "--api-doc-dir", os.path.join(node_path, "api-doc")]
+        invoke_cmd += self._get_node_config_item(i, "extra_args")
+
+        system_log_path = os.path.join(node_path, SCYLLA_LOG_FILE)
+        with open(system_log_path, "w") as system_log:
+            handle = subprocess.Popen(
+                    invoke_cmd,
+                    stdin = subprocess.DEVNULL,
+                    stdout = system_log,
+                    stderr = subprocess.STDOUT,
+                    cwd = os.path.abspath(node_path))
+            self._nodes[i] = {
+                "index": i,
+                "ip": node_ip,
+                "handle": handle,
+            }
+
+        pattern = re.compile("init - Scylla version .* initialization completed.")
+        match = None
+        while match is None:
+            with open(system_log_path, "r") as system_log:
+                for line in system_log:
+                    match = pattern.search(line)
+                    if not match is None:
+                        break
+            time.sleep(0.5)
+
+    def stop_node(self, **kwargs):
+        if self._maybe_apply_all(self.stop_node, **kwargs):
+            return
+        self._logger.info(f"stop_node({kwargs})")
+
+        i = int(kwargs["node_index"])
+
+        if not i in self._config["nodes"]:
```

Comment:
i think, the more idiomatic way in python is just to go ahead, and handle the exception, like
```py

try:
    handle = self._node[i]["handle"]
    handle.terminate()
    handle.wait(timeout=30)
except KeyError:
    return self._make_success(f"Node#{i} is already stopped")
except subprocess.TimeoutExpired:
    handle.kill()
```

Reply:
In general I agree, but here I prefer an explicit check, because there is only one KeyError we can tolerate here.